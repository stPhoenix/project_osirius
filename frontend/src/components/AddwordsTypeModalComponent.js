import React from 'react';
import {Modal, ModalBody, Card, CardBody, CardTitle} from 'reactstrap';


export const AddwordsTypeModalComponent = (props) => {
    return (
        <Modal isOpen={props.modal} toggle={props.toggle_modal} className="p-0 align-self-center">
            <ModalBody className="p-0">
                <Card className="rounded-0">
                    <CardBody className="d-flex flex-column">
                        <CardTitle>Choose category</CardTitle>
                        
                            <form>
                                <select className="custom-form-control" id="learn_language" onChange={props.handleChange} name="category">
                                        {props.cats.map((category) => (<option key={category.name}>{category.name}</option>))}
                                </select>
                            </form>    
                        <button className="btn btn-link align-self-end" onClick={props.add_custom}>Add</button>
                    </CardBody>
                </Card>
            </ModalBody>
        </Modal>
    );
};