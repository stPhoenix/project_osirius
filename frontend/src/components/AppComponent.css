.theme-primary{
    background-color: #009be5;
}
.theme-light{
    background-color: #63ccff;
}
.theme-dark{
    background-color: #006db3;
}