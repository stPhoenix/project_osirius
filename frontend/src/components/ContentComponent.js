import React from 'react';


export const ContentComponent = (props) => {
    return(
        <div className="col-10 d-flex flex-column align-self-center align-items-center justify-content-center p-0">
            {props.children}
        </div>
    );
};