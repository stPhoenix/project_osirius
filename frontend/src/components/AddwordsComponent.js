import React from 'react';
import {Link, Switch, Route} from 'react-router-dom';
import typing from '../assets/typing.png';
import presets from '../assets/presets.png';
import './AppComponent.css';
import AddwordsType from '../containers/AddwordsType';
import AddwordsPresets from '../containers/AddwordsPresets';
import {ContentComponent} from '../components';
import PropTypes from 'prop-types';

const componentProps = {
  location: PropTypes.object.isRequired,  
};

export const AddwordsComponent = ({location}) => {
	if (location.pathname !== "/addwords"){
		return (<Switch>
					<Route path="/addwords/type" component={AddwordsType} />
					<Route path="/addwords/presets" component={AddwordsPresets} />
				</Switch>);	
	};
	return(
    <ContentComponent>
            <section className="d-flex flex-column align-items-center justify-content-center">
		      <div className="d-flex flex-column align-items-center justify-content-center">
                <img src={typing} alt="typing" style={{height:'7rem', width:'auto'}} />
		        <h4 className="opacity-50 text-center px-2 mb-4 mt-0">Type word. System will search word in database or in the internet.</h4>
		        <Link className="custom-btn theme-primary px-3 py-1 mb-5 black" to="/addwords/type" > Add words</Link>
              </div>
              <div className="d-flex flex-column align-top align-items-center justify-content-center">
                <img src={presets} alt="presets" className="" style={{height:'6rem', width:'auto'}} />
		        <h4 className="opacity-50 text-center px-2 mb-4 mt-0">Choose word from preset in database.</h4>
		        <Link className="custom-btn theme-primary px-3 py-1 mb-3 black" to="/addwords/presets" > Add words</Link>
              </div>
            </section>
	</ContentComponent>
		);
};

AddwordsComponent.propTypes = componentProps;