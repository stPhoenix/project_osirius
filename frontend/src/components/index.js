import {AppComponent} from './AppComponent';
import {CustomAlertComponent} from './CustomAlertComponent';
import {LoginComponent} from './LoginComponent';
import {NavBarComponent} from './NavBarComponent';
import {NavItemComponent} from './NavItemComponent';
import {NewsComponent} from './NewsComponent';
import {SignUpComponent} from './SignUpComponent';

export {
    AppComponent,
    CustomAlertComponent,
    LoginComponent,
    NavBarComponent,
    NavItemComponent,
    NewsComponent,
    SignUpComponent
};