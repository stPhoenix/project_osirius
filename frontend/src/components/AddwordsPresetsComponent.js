import React from 'react';
import './AppComponent.css';
import {WordsTableComponent, ContentComponent} from '../components';


export const AddwordsPresetsComponent = (props) => {
    return (<ContentComponent>
                <form className="col-12 col-lg-8 px-0">
                    <label htmlFor="categories" >Choose category:</label>
                    <select className="custom-form-control" id="categories" onChange={props.handleChange} name="category">
                                        {props.cats.map((category) => (<option key={category.name}>{category.name}</option>))}
                    </select>
                </form>
                <div className="d-flex align-self-end mt-3 px-0" >
					<button className="btn btn-link " onClick={props.add_selected}>Add selected</button>
					<button className="btn btn-link pr-0" onClick={props.add_all}>Add all</button>
				</div>
                <WordsTableComponent words={props.words} handleCheck={props.handleCheck} />
          </ContentComponent>);  
};