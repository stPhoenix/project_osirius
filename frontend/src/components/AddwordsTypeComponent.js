import React from 'react';
import './AppComponent.css';
import {CardDeck, Card, CardBody, CardText} from 'reactstrap';
import {AddwordsTypeModalComponent} from '../components';


export const AddwordsTypeComponent = (props) => {
	
	let data = null;
	if (props.global_word_search === true){
		data = (props.words.map(word => {
			return (<Card key={word.id+"words"} className="shadow-sm rounded-0">
                        <CardBody className="d-flex flex-column">
                            <CardText className="text-center">
                                <br />{word.name}
                                <br />{word.pronunciation}
                                <br />Category: [{word.category.map(category => (category.name+" "))}]
                                <br />{word.translation}
                            </CardText>
                            <button name={word.id} className="btn btn-link align-self-end" onClick={props.add_global}>Add</button>
                        </CardBody>
                    </Card>);
		}));
	} else if(props.google_translate_search === true){
		data += (props.words.map(word => {
			return (<Card key={word.id+"words"} className="shadow-sm rounded-0">
                        <CardBody className="d-flex flex-column">
                            <CardText className="text-center">
                                <br />{word.name}
                                <br />{word.pronunciation}
                                <br />{word.translation}
                            </CardText>
                        <button className="btn btn-link align-self-end" onClick={(e) => {props.choose_category(word.name, word.pronuciation, word.translation)}}>Add</button>
                        </CardBody>
                    </Card>);
		}));
	}
	
	return(<div className="col-10 d-flex flex-column px-0 justify-content-center align-items-center">
				<form className="d-flex flex-column w-100">
					<div className="form-group">
						<label htmlFor="search">Type word:</label>
						<input id="search" type="text" className="custom-form-control" value={props.search_word} name="search_word" onChange={props.handleChange}/>
					</div>
                    <button className="custom-btn theme-primary mb-2 align-self-center" onClick={props.search}>Search</button>
				</form>
                <CardDeck className="w-100">
                    {data}
                </CardDeck>
                <AddwordsTypeModalComponent add_custom={props.add_custom}
                                            handleChange={props.handleChange}
                                            cats={props.cats}
                                            modal={props.modal}
                                            toggle_modal={props.toggle_modal} />
		</div>);
};