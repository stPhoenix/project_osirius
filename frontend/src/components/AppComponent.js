import React from 'react';
import {Container, Col, Row} from 'reactstrap';
import './AppComponent.css';
import logo from '../assets/logo.png';
import NavBar from '../containers/NavBar';
import Login from '../containers/Login';
import Alerts from '../containers/Alerts';
import {Switch, Route} from 'react-router';
import News from '../containers/News';
import Logout from '../containers/Logout';
import {SignUpComponent} from '../components';


export const AppComponent = () => {
        return (
      <Container fluid className="bg-first">
        <Row className="d-flex flex-column">
            <Col xs="12" className="px-0" style={{zIndex:1}}>
                <header className="theme-dark px-4 py-2">
                    <img style={{height:'2rem', width:'auto'}} src={logo} alt="logo" />
                </header>
            </Col>
            <Col xs="12" className="px-0" style={{zIndex:1}}>
                <section className="theme-primary shadow-sm full-screen">
                    <NavBar/>
                </section>
            </Col>
            <Col xs="12 px-0">
                <main className="d-flex flex-column full-screen">
                    <Alerts />
                    <section className="d-flex justify-content-center align-items-center">
                        <Switch>
                            <Route exact path="/" component={News} />
                            <Route path="/news" component={News} />
                            <Route path="/login" component={Login} />
                            <Route path="/logout" component={Logout} />
                            <Route path="/signup" component={SignUpComponent} />
                        </Switch>
                    </section>
                    
                </main>
            </Col>
        </Row>
        <Row className="d-flex flex-column">
            <Col xs="12">
                <footer>
                </footer>
            </Col>
        </Row>
      </Container>
    );
  };

