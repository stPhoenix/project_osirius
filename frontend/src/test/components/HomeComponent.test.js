import React from 'react';
import {shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import {HomeComponent} from '../../components';


describe('HomeComponent', () => {
    it('should render correctly', () => {
        const output = shallow(<HomeComponent />);
        expect(shallowToJson(output)).toMatchSnapshot();
    });
});