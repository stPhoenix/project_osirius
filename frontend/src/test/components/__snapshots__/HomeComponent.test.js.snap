// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent should render correctly 1`] = `
<div>
  <section
    className="row d-flex justify-content-center"
  >
    <div
      className="col-12 col-md-11 col-xl-7 my-4"
    >
      <div
        className="row d-flex flex-column flex-lg-row"
      >
        <div
          className="col-12 col-lg-6 col-xl-5"
        >
          <img
            alt="Story"
            className="img-fluid"
            src="story1.png"
          />
        </div>
        <div
          className="col-12 col-lg-6 col-xl-7 my-lg-auto"
        >
          <h2
            className="text-center"
          >
            Hello everyone!
          </h2>
          <p
            className="text-center text-lg-left"
          >
            I've been trying to learn Japanese by myself from textbooks and manuals. When I started I had faced one big problem: how to learn words? The popular solution is to make cards and learn by them. But honestly, for me it was too inconvenient - I'm too lazy to do such things. So I'd found another solution - to write a program that helps in studing Japanese words. And not only Japanese, other languages, too. Don't know why I wasn't lazy to code it)) I keep telling myself that laziness is the engine of progress. Have you already guessed that you're using it right now?
            <br />
            It's all for 
            <strong>
              FREE
            </strong>
            !
          </p>
        </div>
      </div>
    </div>
  </section>
  <section
    className="row bg-second"
  >
    <div
      className="col-12 d-flex mt-2"
    >
      <h2
        className="mx-auto"
      >
        News
      </h2>
    </div>
    <div
      className="col-12 col-md-11 col-xl-7 mx-auto my-3"
    >
      <News />
    </div>
  </section>
  <section
    className="row parallax-bg-2"
  >
    <div
      className="col-12 bg-overlay py-5"
    >
      <div
        className="row d-flex flex-column align-items-center"
      >
        <div
          className="col-12"
        >
          <h2
            className="text-center text-white"
          >
            What app can do for you:
          </h2>
        </div>
        <div
          className="col-12 col-md-11 col-xl-7 my-md-3"
        >
          <div
            className="card-deck"
          >
            <div
              className="card"
            >
              <img
                alt="Learn foreign words"
                className="mx-auto"
                height="auto"
                src="translate_feature.png"
                width="130rem"
              />
              <div
                className="card-body"
              >
                <p
                  className="text-center"
                >
                  Help to learn words in different languages (main idea)
                </p>
              </div>
            </div>
            <div
              className="card"
            >
              <img
                alt="Switch languages feature"
                className="mx-auto"
                height="auto"
                src="switch_feature.png"
                width="130rem"
              />
              <div
                className="card-body"
              >
                <p
                  className="text-center"
                >
                  Switch between different learnt languages
                </p>
              </div>
            </div>
            <div
              className="card"
            >
              <img
                alt="Add words for learning from presets"
                className="mx-auto"
                height="auto"
                src="presets_feature.png"
                width="130rem"
              />
              <div
                className="card-body"
              >
                <p
                  className="text-center"
                >
                  Add words for learning from presets.
                </p>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col-12 col-md-11 col-xl-7 my-md-3"
        >
          <div
            className="card-deck"
          >
            <div
              className="card"
            >
              <img
                alt="Keyboard"
                className="mx-auto"
                height="auto"
                src="keyboard_feature.png"
                width="130rem"
              />
              <div
                className="card-body"
              >
                <p
                  className="text-center"
                >
                  Add words for learning by typing (in this case app will search translation over internet)
                </p>
              </div>
            </div>
            <div
              className="card"
            >
              <img
                alt="Matching"
                className="mx-auto"
                height="auto"
                src="match_feature.png"
                width="130rem"
              />
              <div
                className="card-body"
              >
                <p
                  className="text-center"
                >
                  Study words by matching word with its translation (and reverse)
                </p>
              </div>
            </div>
            <div
              className="card"
            >
              <img
                alt="Book"
                className="mx-auto"
                height="auto"
                src="play_feature.png"
                width="130rem"
              />
              <div
                className="card-body"
              >
                <p
                  className="text-center"
                >
                  Study words by looking at word and typing translation(and reverse)
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
  <Connect(Feedback) />
</div>
`;
